ambry-clustermap/src/main/java/com.github.ambry.clustermap/HelixClusterManager.java

```diff
@@ -155,87 +166,94 @@
         new PartitionSelectionHelper(partitionMap.values(), clusterMapConfig.clusterMapDatacenterName);
   }
 
+  private void setException(boolean isLocalDc, Exception e) {
+    if (isLocalDc) {
+      initializationException.compareAndSet(null, e);
+    } else {
+      remoteInitializationException.compareAndSet(null, e);
+      remoteInitializationFailedCount.incrementAndGet();
+    }
+  }
+
   /**
    * Initialize HelixManager in local datacenter and complete subscription of HelixPropertyStore to listen for
    * PartitionOverride zNode.
    * @param dataCenterToZkAddress the map mapping each datacenter to its corresponding ZkAddress.
    * @param instanceName the String representation of the instance associated with this manager.
    * @param helixFactory the factory class to construct and get a reference to a {@link HelixManager}.
    * @return the HelixManager of local datacenter
-   * @throws IllegalStateException
+   * @throws Exception
    */
   private HelixManager initializeHelixManagerAndPropertyStoreInLocalDC(Map<String, DcZkInfo> dataCenterToZkAddress,
-      String instanceName, HelixFactory helixFactory) throws IllegalStateException {
+      String instanceName, HelixFactory helixFactory) throws Exception {
     DcZkInfo dcZkInfo = dataCenterToZkAddress.get(clusterMapConfig.clusterMapDatacenterName);
     String zkConnectStr = dcZkInfo.getZkConnectStr();
-    HelixManager manager = null;
-    ZkHelixPropertyStore<ZNRecord> helixPropertyStore = null;
-    try {
-      manager = helixFactory.getZKHelixManager(clusterName, instanceName, InstanceType.SPECTATOR, zkConnectStr);
-      logger.info("Connecting to Helix manager in local zookeeper at {}", zkConnectStr);
-      manager.connect();
-      logger.info("Established connection to Helix manager in local zookeeper at {}", zkConnectStr);
-      helixPropertyStore = manager.getHelixPropertyStore();
-      logger.info("HelixPropertyStore from local datacenter {} is: {}", dcZkInfo.getDcName(), helixPropertyStore);
-      IZkDataListener dataListener = new IZkDataListener() {
-        @Override
-        public void handleDataChange(String dataPath, Object data) throws Exception {
-          logger.info("Received data change notification for: {}", dataPath);
-        }
+    HelixManager manager;
+    ZkHelixPropertyStore<ZNRecord> helixPropertyStore;
+    manager = helixFactory.getZKHelixManager(clusterName, instanceName, InstanceType.SPECTATOR, zkConnectStr);
+    logger.info("Connecting to Helix manager in local zookeeper at {}", zkConnectStr);
+    manager.connect();
+    logger.info("Established connection to Helix manager in local zookeeper at {}", zkConnectStr);
+    helixPropertyStore = manager.getHelixPropertyStore();
+    logger.info("HelixPropertyStore from local datacenter {} is: {}", dcZkInfo.getDcName(), helixPropertyStore);
+    IZkDataListener dataListener = new IZkDataListener() {
+      @Override
+      public void handleDataChange(String dataPath, Object data) {
```

Comment:
I think we'd better keep the original method signature. That is keep the `throws Exception`  here. I agree with other code changes in this method.